### ISSUE TITLE
[dv,csr_utils] Fix shadowed CSR partial csr_wr

### ISSUE THREAD

**Issue by matutem** [lowRISC/opentitan#15048]
In clkmgr_base_vseq there is code to enable frequency measurements. In that code we need to update the *_meas_ctrl_shadowed CSRs. Each of these CSRs has a hi and a lo field. If the update is done as two csr_wr, one updating the hi field and another one updating lo we tend to get occasional failures in clkmgr_stress_all_with_rand_reset.

On the other hand, if the register is written in full with the value computed as the hi and lo values masked, shifted, and merged all works well.

**Comment by weicaiyang** [comment]
This could be due to a very corner case on the combination of shadow reg partial write, regwen (gated the shadow reg) and random reset.

Last week, @matutem and I figured out `calib_en` could clear regwen and RAL didn't get updated. That issue was fixed, but the problem isn't completed solved. I guess there may be still another corner issue in this complicated combination. 

Since we @matutem uses another workaround, we could look into this later.
